_posts/2019-01-08-data-driven-static-resume.adoc

```diff
@@ -0,0 +1,202 @@
+[source,yaml]
+----
+include::data/dominick.yml[tags="example-snippet"]
+----
+
+If you inspect the link:https://github.com/briandominick/codewriting/blob/master/data/dominick.yml[whole file], you will see it contains several clusters of data, each analogous to a table in a database.
+
+[TIP]
+One advantage of maintaining small data in flat files is that you can build an entire, perfectly portable datasource with a text editor in mere minutes.
+
+In this case, parameter values contain link:https://asciidoctor.org/docs/what-is-asciidoc/[AsciiDoc] markup.
+Notice on the first line beginning with a `destination:` key, the value text contains AsciiDoc hyperlink markup.
+
+This data gets parsed with a link:https://github.com/Shopify/liquid/wiki[Liquid] template.
+This particular Liquid template (link:https://raw.githubusercontent.com/briandominick/codewriting/master/_templates/liquid/resume.asciidoc[see complete source]) weaves those AsciiDoc-formatted strings from the data file with more AsciiDoc markup.
+
+.From resume.asciidoc -- Template mixes Liquid with AsciiDoc, generates new AsciiDoc source
+[source]
+----
+include::assets/includes/resume.asciidoc[tags="employment"]
+----
+
+Because Liquid templates are usually a mix of at least two markup languages (Liquid markup itself plus your target format), they can get pretty ugly.
+Such is the nature of content templating.
+Nevertheless, we can generate such elegant code this way.
+
+Don't worry about the details of this code.
+Basically what it does is iterate (“loop”) through the `posts.paid` array in the data file, creating an entry for each item it comes across (objects describing my past paid gigs).
+As it churns, Liquid expresses the details of each object, mixing it in with some AsciiDoc markup you'll also recognize in the output just below.
+
+.From _build/pages/resume.adoc -- AsciiDoc source file output after parsing
+[source,asciidoc]
+----
+include::pages/resume.adoc[tags="example-snippet"]
+----
+
+The above snippet includes just the AsciiDoc generated for the first two entries, though the `_build/pages/resume.adoc` file is a complete AsciiDoc document source.
+
+[NOTE]
+Since this file is built and thus not tracked in the Git repository, to see the whole file you would have to clone and build the Codewriting repo, generating a copy into the `_build/pages/` directory.)
+
+[TIP]
+Because that final, prebuilt source is generated before this blog, the above snippet is directly sourced from the resulting file, accessed while this blog post was being rendered and ensuring it always accurately reflects the _current_ state of the underlying data.
+(See the <<sniptip,bonus explainer>> at the end of this post.)
+
+Finally, I render that AsciiDoc file into rich-text output.
+The PDF document is generated by the killer link:http://prawnpdf.org/api-docs/2.0/[*Prawn* PDF utility], after which the link:https://jekyllrb.com/[Jekyll static site generator] spits it out as HTML.
```

Comment:
As a reader, I am wondering why PDF comes before HTML. You might consider briefly addressing the 'why' of this here.